src/librustc_trans/back/linker.rs

```diff
@@ -199,6 +201,37 @@ impl<'a> Linker for GnuLinker<'a> {
     fn export_symbols(&mut self, _: &Session, _: &CrateTranslation, _: &Path) {
         // noop, visibility in object files takes care of this
     }
+
+    fn try_gold_linker(&mut self) {
+        if !cfg!(target_os = "linux") {
```

Comment:
No I didn't. I'm reluctant to impose this on them without proof it works though. We have lots of visibility into Linux, little into the BSDs. If a BSD maintainer wants gold, perhaps they can submit a follow up verifying that it works.